test/spoof/spoof.go

```diff
@@ -174,3 +189,33 @@ func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker) (*Res
 
 	return resp, err
 }
+
+// LogZipkinTrace provides support to log Zipkin Trace for param: traceID
+func (sc *SpoofingClient) LogZipkinTrace(traceID string) error {
+
+	if zipkin.CheckZipkinPortAvailability() {
+		return errors.New("Zipkin Port(9411) is not-setup. failing Zipkin Trace retrieval")
+	}
+
+	sc.logger.Infof("Logging Zipkin Trace: %s", traceID)
+
+	zipkinTraceEndpoint := fmt.Sprintf(zipkin.ZipkinTraceEndpoint, traceID)
+
+	// Sleep to ensure all traces are correctly pushed on the backend.
+	time.Sleep(5 * time.Second)
+	resp, err := http.Get(zipkinTraceEndpoint)
+
+	if err != nil {
+		return errors.New("Error retrieving Zipkin trace")
```

Comment:
Augment the existing error message instead of providing a generic one without details.

Reply:
Fixed.